ls1intum/Artemis
# `General`: Fix exercise statistics chart not being able to load data from the server

### Checklist
#### General


#### Server

### Motivation and Context
In the student exercise overview page, there is a tab with statistics. There, the last chart ‘Performance in Exercises’ could not load any data due to an internal server error.

### Description
The implementation used `parallelStream` to filter the list of exercises for the student. However, the authentication information is only present in the thread that is responsible for answering the request. If the parallel implementation makes a database call in a different thread, then it fails. Replacing `parallelStream` with a regular `stream` fixes this issue.

I also refactored the resource to remove an unnecessary nested class.

### Steps for Testing
Prerequisites:
- 1 Student

1. Log in to Artemis
2. Go to a course.
3. Go to the ‘Statistics‘ tab. No error alert should be shown, all charts are rendered.

### Review Progress





#### Code Review
#### Manual Tests

### Test Coverage
unchanged
